### PR TITLE
Fix dependency versions for net8 target

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Addresses https://github.com/dotnet/SqlClient/pull/3063#discussion_r1874029699

### CODE REVIEW

**Review by David-Engel — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs [hunk 1]
```diff
@@ -210,8 +218,11 @@ internal static bool VerifyRsaSignatureDirectly(byte[] hashedCek, byte[] signedC
         {
             Debug.Assert(rsaPfx != null && rsaPfx.Length > 0);
 
+#if NET9_0_OR_GREATER
             X509Certificate2 x509 = X509CertificateLoader.LoadPkcs12(rsaPfx, @"P@zzw0rD!SqlvN3x+");
-
+#else
+            X509Certificate2 x509 = new(rsaPfx, @"P@zzw0rD!SqlvN3x+");
```
[the remark this answers is not in the document]
Comment (edwardneal) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs:224]: There are quite a few of these. I've already removed them from the functional tests, and I'm thinking about the best way to remove them from the manual tests.

File: tools/props/VersionsNet9OrLater.props [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (cheenamalhotra) [reply tools/props/VersionsNet9OrLater.props:1]: I can't recall where was the discussion.. we need this for the reason @David-Engel mentioned in the comment linked in PR description.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs:224]: I don't think these are scanned as secrets so far. Cleanup definitely needs to be done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs [hunk 3]
```diff
@@ -201,7 +201,7 @@ internal static byte[] DecryptRsaDirectly(byte[] rsaPfx, byte[] ciphertextCek, s
             Debug.Assert(rsaPfx != null && rsaPfx.Length > 0);
             // The rest of the parameters may be invalid for exception handling test cases
 
-#if NET9_0_OR_GREATER
+#if NET9_0
```
Comment (mdaigle) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs:204]: How did these changes fix the ARM builds? Are the constants defined differently there?

**Review by mdaigle — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs (hunk 3 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtility.cs:204]: It seems the NET9_0_OR_GREATER constant isn't available everywhere as of now. I found that missing in the logs where the build failed.

**Pull request merged**